Cabal/Distribution/Types/LocalBuildInfo.hs

```diff
@@ -301,3 +308,15 @@ externalPackageDeps lbi =
     -- True if this dependency is an internal one (depends on the library
     -- defined in the same package).
     internal ipkgid = any ((==ipkgid) . componentUnitId) (Graph.toList (componentGraph lbi))
+
+withProfLib :: LocalBuildInfo -> Bool
+withProfLib = isJust . libProfiling
```

Comment:
@ezyang I was on the fence about renaming these accessors anyway because the compiler-specific build logic is very tightly integrated with the access pattern. I will just revert this part of the patch.